src/qmapshack/gis/rte/router/CRouterBRouter.cpp

```diff
@@ -383,9 +383,12 @@ int CRouterBRouter::synchronousRequest(const QVector<QPointF>& points, const QLi
   } catch (const QString& msg) {
     coords.clear();
     if (!msg.isEmpty()) {
+      CCanvas* canvas = CMainWindow::self().getVisibleCanvas();
+      if (canvas) {
+        canvas->reportStatus("BRouter", tr("<b>BRouter</b><br/>Bad response from server:<br/>%1").arg(msg));
```

Comment:
this status-message cannot be seen as it is closed immediately by the call of 'slotCloseStatusMsg()' in line 396

Reply:
You got a point here.  In the meanwhile I could follow the call paths and I am pretty confident that the exceptions are caught while routing on-the-fly. 

I added a few `noexcept(false)`  for readability. Else they are worth nothing.